ocs_ci/ocs/platform_nodes.py

```diff
@@ -941,3 +941,121 @@ def get_worker_resource_id(self, resource):
+        self.baremetal = baremetal.BAREMETAL()
+
+    def stop_nodes(self, nodes, force=True):
+        """
+        Stop Baremetal Machine
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            force (bool): True
+
+        """
+        self.baremetal.stop_baremetal_machines(nodes, force=force)
+
+    def start_nodes(self, nodes, wait=True):
+        """
+        Start Baremetal Machine
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): Wait for node status
+
+        """
+        self.baremetal.start_baremetal_machines(nodes, wait=wait)
+
+    def restart_nodes(self, nodes, force=True):
+        """
+        Restart Baremetal Machine
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            force (bool): True for force VM stop, False otherwise
+
+        """
+        self.baremetal.restart_baremetal_machines(nodes, force=force)
+
+    def restart_nodes_teardown(self):
+        """
+        Make sure all VMs are up by the end of the test
+
+        """
+        bms = self.baremetal.get_nodes_ipmi_ctx(self.cluster_nodes)
+        stopped_bms = [
+            bm for bm in bms if self.baremetal.get_power_status(bm) == constants.VM_POWERED_OFF
+        ]
+
+        if stopped_bms:
+            logger.info(f"The following BMs are powered off: {stopped_bms}")
+            self.baremetal.start_baremetal_machines_with_ipmi_ctx(stopped_bms)
+        for bm in bms:
+            bm.session.close()
+
+    def get_data_volumes(self):
+        raise NotImplementedError(
+            "Get data volume functionality is not implemented"
+        )
+
+    def get_node_by_attached_volume(self, volume):
+        raise NotImplementedError(
+            "Get node by attached volume functionality is not implemented"
+        )
+
+    def detach_volume(self, volume, node=None, delete_from_backend=True):
+        raise NotImplementedError(
+            "Detach volume functionality is not implemented"
+        )
+
+    def attach_volume(self, volume, node):
+        raise NotImplementedError(
+            "Attach volume functionality is not implemented"
+        )
+
+    def wait_for_volume_attach(self, volume):
+        raise NotImplementedError(
+            "Wait for volume attach functionality is not implemented"
+        )
+
+    def create_and_attach_nodes_to_cluster(self, node_conf, node_type, num_nodes):
+        raise NotImplementedError(
+            "attach nodes to cluster functionality is not implemented"
+        )
+
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        raise NotImplementedError(
+            "attach nodes to cluster functionality is not implemented"
+        )
+
+    def attach_nodes_to_cluster(self, node_list):
+        raise NotImplementedError(
+            "attach nodes to cluster functionality is not implemented"
+        )
+
+    def read_default_config(self, default_config_path):
```

Comment:
I think this method needs to be added also to the base class (NodesBase)

Reply:
It's already available in NodesBase

Reply:
correct. Missed it, I thought it does not exist